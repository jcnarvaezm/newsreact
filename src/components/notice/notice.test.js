// import React from 'react';
// import { configure, shallow } from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
// import Notice from './notice';

// configure({
//   adapter: new Adapter(),
// });

// describe('Testing Notice Component', () => {
//   test('Testing rendering of Notice Component', () => {
//     const wrapper = shallow(<Notice />);
//     expect(wrapper.find('section').exists()).toBe(true);
//   });
// });
import React from 'react';
import { create, act } from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Notice from './notice';

configure({
  adapter: new Adapter(),
});

describe('Testing Form Container', () => {
  let component;
  let tree;
  let instance;
  let content;

  const event = {
    preventDefault: jest.fn(),
  };

  const handleDismissMock = jest.fn();

  beforeAll(() => {
    component = create(<Notice />);
    tree = component.toJSON();
    instance = component.root;
  });

  test('renders form container correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  test('Content has class', () => {
    content = instance.findByType('div');
    expect(content.props.className).toEqual('eui-alert eui-alert-info');
  });
  test('', () => {
    let form = instance.findByType('button');
    // Submit form
    act(() => {
      form.props.onClick(event);
    });
    expect(handleDismissMock).toHaveBeenCalledTimes(0);
  });
});
