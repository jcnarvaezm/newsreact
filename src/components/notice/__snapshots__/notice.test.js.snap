// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Form Container renders form container correctly 1`] = `
<div
  className="eui-alert eui-alert-info"
  role="alert"
>
  <section>
    Welcome to the new look of News.com. Keep scrolling to discover interesting new features and news.
  </section>
  <button
    className="eui-icon-button"
    disabled={false}
    onClick={[Function]}
  >
    <i
      className="eui-icon material-icons"
    >
      cancel
    </i>
  </button>
</div>
`;
